### PR TITLE
[WIP] Synchronize key-value pairs across workers

### PR DESCRIPTION
**Pull request by dharmit**

### CODE REVIEW

File: container_pipeline/pipeline.py [hunk 1]
```diff
@@ -3,60 +3,137 @@
...
-    job_name = utils.get_job_name({
-        'appid': appid, 'jobid': jobid, 'desired_tag': desired_tag})
-    project = utils.get_job_hash(job_name)
+def create_new_job():
+    """
+    Creates new job (a dictionary) with default value as `None` for all the
+    keys. This is the central place for creation of a job for container
+    pipeline service
+    """
+    job = dict.fromkeys([
```
Comment (bamachrn) [inline container_pipeline/pipeline.py]: grouping the similar/related keys would be better.

File: container_pipeline/workers/build.py [hunk 2]
```diff
@@ -27,14 +27,16 @@ def handle_job(self, job):
         queue to be processed later. Else, it goes ahead with running
         build for the job.
         """
+        self.job = job
```
Comment (bamachrn) [inline container_pipeline/workers/build.py]: this wont break the inheritance right? I mean base worker calls the handle build job from run method.

File: container_pipeline/workers/test.py [hunk 3]
```diff
@@ -45,56 +45,58 @@ def run_test(self, job):
...
+        self.job['output_image'] = \
+            "registry.centos.org/{}/{}:{}".format(
+                self.job['appid'], self.job['jobid'], self.job['desired_tag'])
+        self.job['beanstalk_server'] = settings.BEANSTALKD_HOST
+        self.job['image_name'] = \
+            "{}/{}:{}".format(
+                self.job['appid'], self.job['jobid'], self.job['desired_tag'])
+
+        self.queue.put(json.dumps(self.job), 'master_tube')
         self.logger.debug("Test is successful going for next job")
```
Comment (bamachrn) [inline container_pipeline/workers/test.py]: if we can set all this in linter worker or in the job initialization only. IMO, setting value for parameter in the worker should be there only that worker is changing the value. and these are no change

File: container_pipeline/pipeline.py (hunk 1 above)
Comment (dharmit) [reply container_pipeline/pipeline.py]: how about making it chronological to make searching/accessing easier? Like this:
```python
job = dict.fromkeys(['action',
             'appid',
             'beanstalk_server',
             'build_status',
             'delivery_status',
             'depends_on',
             'desired_tag',
             'dockerfile',
             'image_name',
             'image_under_test',
             'job_name',
             'jobid',
             'last_run_timestamp',
             'lint_status',
             'logs_URL',
             'logs_dir',
             'logs_file_path',
             'namespace',
             'notify_email',
             'output_image',
             'project_hash_key',
             'project_name',
             'repo_branch',
             'repo_build_path',
             'repo_url',
             'retry',
             'retry_delay',
             'scan_status',
             'target_file',
             'test_tag'])
```

File: container_pipeline/workers/build.py (hunk 2 above)
Comment (dharmit) [reply container_pipeline/workers/build.py]: No this won't break inheritance. Base worker calls `handle_job` function with a parameter called `job` while we're assigning its value to a instance variable which has value `None`. In fact, I'm planning to move `self.job = None` from workers' `__init__` method to the base class's `__init__` method.

File: container_pipeline/workers/test.py (hunk 3 above)
Comment (dharmit) [reply container_pipeline/workers/test.py]: That's the anomaly I was talking about. Clearly I missed it. Thank you!

**Comment by dharmit** [comment]
Closing this as I just needed preliminary feedback. Will open a new PR once the task has been accomplished for all workers.

**Pull request closed without merging**